lib.ts

```diff
@@ -72,12 +74,26 @@ function dataUriToBuffer(uri: string): MimeBuffer {
 export type InitInput = RequestInfo | URL | Response | BufferSource | WebAssembly.Module;
 
 let inited: Promise<any> | null = null;
-export const init = async (input?: InitInput | Promise<InitInput>, force?: boolean) => {
+
+/**
+ * Initialize the underlying WebAssembly module.
+*/
+export const init = async (input?: InitInput | Promise<InitInput>, maybe_memory?: WebAssembly.Memory, force?: boolean) => {
     if (inited === null || force === true) {
 		if (!input) {
 			input = await WebAssembly.compile(dataUriToBuffer(wasm_bytes as any as string));
 		}
-        inited = load(input);
+        inited = load(input, maybe_memory);
     }
     await inited;
 }
+
+// HACK: We save these to the global scope because it's the most reliable way to
+// make sure worker.js gets access to them. Normal exports are removed when
+// using a bundler.
+(globalThis as any)["__WASMER_INTERNALS__"] = { ThreadPoolWorker, init };
+
+// HACK: Temporary polyfill so console.log() will work with BigInt.
+(BigInt as any).prototype.toJSON = function () {
+	return this.toString();
+  };
```

Comment:
Interesting. if we read it from JSON, it would be converted to BigInt or to a String? (will this cause any issue?)

Reply:
That's just a hack for now so `console.log()` can print the messages we're sending back and forth between workers (Rust pointers are serialized as `BigInt`). We don't actually use `JSON.stringify()` at the moment because all messages are true JavaScript objects.

Once we're happy with how messages are being sent around, we can remove those lines.

Reply:
Sounds good